packages/flutter/lib/src/widgets/form.dart

```diff
@@ -139,13 +155,18 @@ class Form extends StatefulWidget {
 /// Typically obtained via [Form.of].
 class FormState extends State<Form> {
   int _generation = 0;
+  bool _hasInteractedByUser = false;
   final Set<FormFieldState<dynamic>> _fields = <FormFieldState<dynamic>>{};
 
   // Called when a form field has changed. This will cause all form fields
   // to rebuild, useful if form fields have interdependencies.
   void _fieldDidChange() {
     if (widget.onChanged != null)
       widget.onChanged();
+    if (!_hasInteractedByUser) {
+      _hasInteractedByUser = _fields
```

Comment:
This may be nit picking, but what if the _hasInteractedByUser is set to true due to one field has change, and later on the field get reseted? I believe the form will still autovalidating in this case. This may be an ok behavior, but we should mention it in the documentation.

Reply:
Good catch!

Reply:
Wow, lets just revert this for now.

Reply:
instead of keeping track of a bool, can we keep track the number of field that is dirty? when ever a field become dirty or not dirty, we update that number  +1 or -1. The auto validate in the form will just be number>0

Reply:
Or could _fieldDidChange receive some extra information to make this easier?  Like maybe the field(s) that changed, or maybe a bool indicating if it was a reset or not.

Reply:
> Or could _fieldDidChange receive some extra information to make this easier? Like maybe the field(s) that changed, or maybe a bool indicating if it was a reset or not.

I love the _fieldDidChange receiving extra information approach because with this will be easy to extend this feature with validating only the field that has changed in the future (I mean if necessary).